pedsnetdcc/prepdb.py

```diff
@@ -0,0 +1,244 @@
+grant all    on               schema {{.Site}}_id_maps  to loading_user;
+alter default privileges for role loading_user in schema {{.Site}}_pedsnet grant select on tables to harvest_user, achilles_user, dqa_user, pcor_et_user, peds_staff;
+alter default privileges for role loading_user in schema {{.Site}}_pcornet grant select on tables to peds_staff;
+"""
+
+# SQL template for creating site schemas in a production database instance.
+_sql_site_prod_template = """
+create schema if not exists {{.Site}}_pedsnet authorization dcc_owner;
+create schema if not exists {{.Site}}_pcornet authorization dcc_owner;
+create schema if not exists {{.Site}}_harvest authorization dcc_owner;
+create schema if not exists {{.Site}}_id_maps  authorization dcc_owner;
+grant usage  on               schema {{.Site}}_pedsnet  to harvest_user, peds_staff;
+grant select on all tables in schema {{.Site}}_pedsnet  to harvest_user, peds_staff;
+grant usage  on               schema {{.Site}}_pcornet  to pcornet_sas, peds_staff;
+grant select on all tables in schema {{.Site}}_pcornet  to pcornet_sas, peds_staff;
+grant all    on               schema {{.Site}}_harvest  to harvest_user;
+grant all    on               schema {{.Site}}_id_maps  to loading_user;
+alter default privileges for role dcc_owner in schema {{.Site}}_pedsnet grant select on tables to harvest_user, peds_staff;
+alter default privileges for role dcc_owner in schema {{.Site}}_pcornet grant select on tables to pcornet_sas, peds_staff;
+"""
+
+
+def _site_sql(site, db_type):
+    """Return a list of statements to create the schemas for a given site.
+
+    :param site: site name, e.g. 'dcc' or 'stlouis'
+    :type: str
+    :param db_type: 'prod' or 'internal'
+    :type: str
+    :return: SQL statements
+    :rtype: list(str)
+    :raises: ValueError
+    """
+    if db_type == 'internal':
+        tmpl = _sql_site_internal_template
+    elif db_type == 'prod':
+        tmpl = _sql_site_prod_template
+    else:
+        raise ValueError('Invalid db_type: {}'.format(db_type))
+
+    sql = tmpl.replace('{{.Site}}', site)
+    return [x for x in sql.split("\n") if x]
+
+
+_sql_vocab_internal = """
+grant all                  on schema vocabulary to loading_user;
+grant usage                on schema vocabulary to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+grant select on all tables in schema vocabulary to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+alter default privileges for role loading_user in schema vocabulary grant select on tables to achilles_user, dqa_user, pcor_et_user, harvest_user, peds_staff;
+"""
+
+_sql_vocab_prod = """
+grant usage                on schema vocabulary to harvest_user, peds_staff;
+grant select on all tables in schema vocabulary to harvest_user, peds_staff;
+alter default privileges for role dcc_owner in schema vocabulary grant select on tables to harvest_user, peds_staff;
```

Comment:
That was the original idea, that `loading_user` wouldn't even exist in prod -- only the users actually necessary.  However, I think at some point I swung around and decided that it would be easier to just have all users created in both databases. Then we can do vanilla dumps and restores and have just one source of users to maintain, and hopefuly one set of privileges.